src/serial.rs

```diff
@@ -52,9 +77,14 @@ use crate::gpio::gpiob::{PB10, PB11, PB6, PB7};
 use crate::gpio::gpioc::{PC10, PC11};
 use crate::gpio::gpiod::{PD5, PD6, PD8, PD9};
 use crate::gpio::{Alternate, Floating, Input, PushPull};
-use crate::rcc::{RccBus, Clocks, Enable, Reset, GetBusFreq};
+use crate::rcc::{RccBus, Clocks, GetBusFreq, sealed::{Enable, Reset}};
 use crate::time::{U32Ext, Bps};
 
+pub trait RegisterBlock {
+    type RB;
+    unsafe fn rb() -> *const Self::RB;
```

Comment:
like this? :)

Reply:
Yep, and without `Deref` bounds in the code.

Reply:
Note that the method is not unsafe, only dereferencing the returned pointer is unsafe, so no need of unsafe in this trait.

Reply:
Ok. If this trait will common, where (what module) it is better to locate it?

Reply:
it should go in stm32hal-utils crate ;-)

Waiting for such a project, I'd say a `pub(crate) mod utils;` at the top level.

Reply:
Can I ask question? Why you don't like `Deref`?

Reply:
See https://github.com/stm32-rs/stm32f1xx-hal/pull/161#issuecomment-570601881

Reply:
Moreover, the deref implementation will be available to the user, and, for me, it looks more like an implementation detail than a functionnality.

Reply:
Done.